src/WebJobs.Script/Description/FunctionDescriptorProvider.cs

```diff
@@ -200,6 +193,20 @@ protected internal virtual void ValidateFunction(FunctionMetadata functionMetada
                     names.Add(binding.Name);
                 }
             }
+
+            // Verify there aren't multiple triggers defined
+            var triggers = functionMetadata.InputBindings.Where(p => p.IsTrigger).ToArray();
+            if (triggers.Length > 1)
+            {
+                throw new InvalidOperationException($"Multiple trigger bindings defined. A function can only have a single trigger binding.");
```

Comment:
To get a proper error message, we need to do this validation here, otherwise we get a "sequence contains more than one matching element" [here](https://github.com/Azure/azure-functions-host/blob/8596a27c5a3cf1c26546418fe8ac6bcf132f808b/src/WebJobs.Script/Description/FunctionDescriptor.cs#L39). So we need to ensure we never create an invalid FunctionDescriptor in the first place.